app/services/visit_request/cancel.rb

```diff
@@ -6,10 +6,23 @@ def initialize(visit_request)
 
     def call
       visit_request.canceled!
+      attend_first_user_from_waitlist
     end
 
     private
 
     attr_reader :visit_request
+
+    def attend_first_user_from_waitlist
+      if waiting_list_request = VisitRequest.where(waiting_list: true).first
+        waiting_list_request.waiting_list = false
```

Comment:
You can replace 
```
waiting_list_request.waiting_list = false
waiting_list_request.save
``` 
with one line:
`waiting_list_request.main_list!` [defined here](https://github.com/pivorakmeetup/pivorak-web-app/blob/development/app/models/visit_request.rb#L24)